### PR TITLE
Add `GetWithdrawalEstimation`

### PR DESCRIPTION
**Pull request by syjn99**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

```bash
GET /over/v1/beacon/states/{state_id}/withdrawal_estimation/{validator_id}
```

Add API for deposit estimation for given `state_id` and `validator_id`.

Here is a schema for the response:

```
// Response
{
	"data": {
		"pubkey": <string>, // Returns the validator pubkey as is
		"pending_partial_withdrawals": List[PendingPartialWithdrawals] // For the given state_id and validator_id, returns any pending partial withdrawal requests with their expected values.
	}
}

// PendingPartialWithdrawals
{
	"data": {
		"amount": <string>,
		"expected_epoch": <string>, // Predicted epoch when this partial withdrawal will be processed (EL add balance).
	}
}
```

**Other notes for review**
- Suggestion for test cases is welcomed.
- `handlers_withdrawal_test.go` includes several scenarios that might help you to understand.

### CODE REVIEW

File: beacon-chain/rpc/over/handlers_withdrawal_test.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+				PendingPartialWithdrawals: []*structs.PendingPartialWithdrawalContainer{
+					{
+						Amount:        100_000_000_000,                                                    // 100 OVER
+						ExpectedEpoch: uint64(params.BeaconConfig().MinValidatorWithdrawabilityDelay) + 1, // 1 epoch later
+					},
+				},
+			},
+		},
+		{
+			name:        "expected epoch is larger because of max pending partial withdrawals per sweep",
```
Comment (schyun9212) [inline beacon-chain/rpc/over/handlers_withdrawal_test.go]: It seems like the same test is being performed redundantly.

**Review by schyun9212 — COMMENTED** [review]
The logic seems fine, but if there are a large number of pending withdrawals accumulated in the queue, this could become too resource-intensive.

It would be fine to consider adding an option to the API to set a limit on the number of items to search or restrict the range of epochs to explore.

File: beacon-chain/rpc/over/handlers_withdrawal_test.go (hunk 1 above)
Comment (syjn99) [reply beacon-chain/rpc/over/handlers_withdrawal_test.go]: Deleted.

**Comment by syjn99** [comment]
@schyun9212 

> It would be fine to consider adding an option to the API to set a limit on the number of items to search or restrict the range of epochs to explore.

Thanks, I will consider adding query params for limiting the iteration.

**Review by schyun9212 — APPROVED** [review]
With the addition of the limit option, there may be cases where withdrawals are created but cannot be viewed via the API.

It would be good to be aware of this aspect moving forward.

**Pull request merged**